pkg/providers/tf/provider.go

```
@@ -132,7 +133,7 @@ func (provider *TerraformProvider) destroy(ctx context.Context, deploymentID str
 	}
 
 	go func() {
-		err = provider.performTerraformUpgrade(ctx, workspace)
+		err = provider.checkTerraformVersion(ctx, workspace)
```

I think this check should not be be in the goroutine. The check should be done as soon as we get the workspace and fail fast as part of the initial response, not as a last operation failure.
There is another question in my mind about what error should be returned. I think the message should mention a tip to run upgrade-service, but possible also respond with a `MaintenanceInfoConflict` error

---

Yeah - in my mind this would all be done as part of https://www.pivotaltracker.com/story/show/182208954. This chore was simply stripping out the upgrade behaviour and leaving the rest of the current logic untouched, with a view that it will all be solved in that story.